crates/op-rbuilder/src/tests/framework/harness.rs

```diff
@@ -190,8 +190,72 @@ impl TestHarness {
         BUILDER_PRIVATE_KEY
     }
 
-    pub const fn builder_log_path(&self) -> &PathBuf {
-        &self.builder_log_path
+    pub async fn check_tx_in_pool(&self, tx_hash: B256) -> eyre::Result<TransactionStatus> {
+        let pool_inspect = self
+            .provider()
+            .expect("provider not available")
+            .txpool_content()
+            .await?;
+
+        let is_pending = pool_inspect.pending.iter().any(|pending_account_map| {
+            pending_account_map
+                .1
+                .iter()
+                .any(|(_, tx)| tx.as_recovered().hash() == *tx_hash)
+        });
+        if is_pending {
+            return Ok(TransactionStatus::Pending);
+        }
+
+        let is_queued = pool_inspect.queued.iter().any(|queued_account_map| {
+            queued_account_map
+                .1
+                .iter()
+                .any(|(_, tx)| tx.as_recovered().hash() == *tx_hash)
+        });
+        if is_queued {
+            return Ok(TransactionStatus::Queued);
+        }
+
+        // check that the builder emitted logs for the reverted transactions with the monitoring logic
+        // this will tell us whether the builder dropped the transaction
+        // TODO: this is not ideal, lets find a different way to detect this
+        // Each time a transaction is dropped, it emits a log like this
+        // Note that this does not tell us the reason why the transaction was dropped. Ideally
+        // we should know it at this point.
+        // 'Transaction event received target="monitoring" tx_hash="<tx_hash>" kind="discarded"'
+        let builder_logs = std::fs::read_to_string(&self.builder_log_path)?;
+        let txn_log = format!(
+            "Transaction event received target=\"monitoring\" tx_hash=\"{}\" kind=\"discarded\"",
```

Comment:
Def not within the scope of this PR, but in the future I think that we should make this more type safe and insert a `#[cfg(test)] debug_probe: DebugProbe` somewhere in the builder to verify those types of things more reliably. Checking logs texts can be very fragile in the long run.